src/osdc/ObjectCacher.cc

```diff
@@ -1276,13 +1276,13 @@ void ObjectCacher::bh_write_commit(int64_t poolid, sobject_t oid,
     finish_contexts(cct, ls, r);
 }
 
-void ObjectCacher::flush(ZTracer::Trace *trace, loff_t amount)
+void ObjectCacher::flush(ZTracer::Trace *trace, loff_t amount, loff_t max_bhs)
```

Comment:
If it's a count, I'd prefer an unsigned signature, even if internally it's converted to an int (in which case the conversion should be validated). `bh_write_adjacencies()` could be refactored to take something unsigned, too, if it doesn't require the value to be signed. `size_t` would be my first choice

Reply:
```
void ObjectCacher::bh_write_adjacencies(BufferHead *bh, ceph::real_time cutoff,
					int64_t *max_amount, int *max_count)
```

this signature could be better with `uint64_t *max_amount, size_t *max_count`

Reply:
Since this is a bug fix that (I assume) is going to be backported, the strategy should be fitting into existing code as much as possible.  Changing a signature of a method that is otherwise isn't touched, which means unnecessarily touching all of its callers, is better left for a separate cleanup PR.